gnovm/tests/files/const9.gno

```diff
@@ -14,4 +14,4 @@ func main() {
 }
 
 // Error:
-// main/files/const9.gno:1: constant definition loop with b
```

Comment:
This is incorrect: it's considering :3 as the location instead of `c` (or wherever else this happens)

Reply:
It is considered line 3 as the starting point for the declaration statement that begins with `const (` and ends at line 10. I will investigate if we can pinpoint the exact assignment line with the current design.

Reply:
It is possible to replicate the cycle lines like Golang's as follows (to make the error more accurate):

```
./prog.go:5:2: initialization cycle for b
	./prog.go:5:2: b is referred to
	./prog.go:6:2: c is referred to
	./prog.go:7:2: d is referred to
	./prog.go:8:2: e is referred to
	./prog.go:5:2: b
```
However, some refactoring is required in the functions [predefinenow](https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3001) and [trydefine](https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3154) to make this error more precise.

Should we implement these changes in this PR?

Reply:
I guess you want introduce a dependency graph for this, but i’m not sure we really should do it. had a similar talk in #2081.

Reply:
I plan to use the existing check circularity variable at https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L3036-L3039. If we encounter the first circularity, I will handle the panic situation differently I  want to  display all variables concerned.

Reply:
@omarsy please, tackle this in another PR: the issue is not circularity, but where we mark the error as happening. the location should happen at the individual declaration, not at the start of the declaration block. thanks.

Reply:
@thehowl, is this one good for you?
Commit: https://github.com/gnolang/gno/pull/2221/commits/eaab6e11e1eea0b39a2f84ac3dcad20afe7f4429 ?
I no longer use the line from the declaration group.